### PR TITLE
Milestone/Project 4: UI 🎫

### PR DESCRIPTION
**Pull request by ITurres**
# 🚩I have made the following changes to complete OOP School Library - Milestone/Project 4

### Type of **`files`** changed/added

My Skills

### In this new Branch, I have **➕ADDED➕** the succeeding **\*📂 folders** and/or **\*\*📄 files** to complete this milestone's [requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_ui.md)

# \*📂 root/

- ## \*📂 helpers/

  - **\*\*📄 run_options.rb** ➕

    - This helper is used to run the program options to the console and get the user input
      for the option to run by `app.rb`.

  - **\*\*📄 options_hash.rb** ➕

    - This method returns an object with all the option's `name` and `methods` to run the program.

  - **\*\*📄 validate_inputs.rb** ➕

    - This helper is used to validate the `age` and `name` inputs, checking if they are `Integer` and `String` respectively.

- ## \*📂 ui/

  - **\*\*📄 app.rb** ➕

    - The **`-App-`** class will be the console app entry point.

      - It contains the following methods:

        - `list_all_books`
        - `list_all_people`
        - `create_teacher_or_student`
        - `create_book`
        - `create_book_rental`
        - `list_all_rentals`
        - `exit_program`

  - **\*\*📄 dependencies.rb** ➕

    - The dependencies file is used to require all the files needed to run the program.
    - This will help to avoid requiring files in the `main.rb` file or elsewhere.

  - **\*\*📄 main.rb** ➕

    - The `main` method holds the only instance of the **`-App-`** class gets the options to run the program and invokes `run_options` passing the options and the app instance to run the program.
    - This method is invoked right after its definition.

### I have **✏️MODIFIED✏️** the succeeding **\*📂 folders** and/or **📄\*\* files**

# \*📂 root/

- ## \*📂 lib/

  - **\*\*📄 classroom.rb** ✏️
  - **\*\*📄 book.rb** ✏️
  - **\*\*📄 rental.rb** ✏️

    - I have fixed the typo on the metaprogramming setter/getter from `attr_assessor` to `attr_accessor`.

  - **\*\*📄 teacher.rb** ✏️

    - Now **`-Teacher-`** will inherit from **`-Person-`** as it should have done from the beginning of its creation.
    - I've modified attr_reader to attr_accessor.  

---

### Thank you for taking the time to review this PR ⭐

---

If you need additional information or have any questions, please don't hesitate to contact me on Slack as _**Arturo (Arthur) Emanuel Guerra Iturres**_. I'll be happy to help you out. 🎯

---

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ Example commands:
 
 - [x] Class Decorators.
 - [x] Associations.
-- [ ] Basic UI.
+- [x] Basic UI.
 - [ ] Preserve data.
```
Comment (Deepakdanger) [inline README.md:140]: - [x] I would recommend adding the steps on how to run the application after cloning. Please add the steps in the README.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @ITurres, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Application are running as expected.

## Required changes
_Check the comments under the review._


## Optional changes
You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 1 above)
Comment (ITurres) [reply README.md:140]: Hi @Deepakdanger! Absolutely, I'll make the changes right away 🎯. Thank you!

**Review by Efiamotu-1 — APPROVED** [review]
Hi @ITurres, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# STATUS: APPROVED 🕺🥳 

image

## Highlights :dart: 
* Correct Gitflow :heavy_check_mark: 
* Good commit messages :heavy_check_mark: 
* All linters passing :heavy_check_mark: 
* Good PR summary :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, you can now Merge since it is Approved ✅, However if there are any conflicts please resolve them before merging.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**